client/lib/signup/step-actions.js

```diff
@@ -144,11 +151,17 @@ export function createSiteWithCart(
 			// step object itself depending on if the theme is provided in a
 			// query. See `getThemeSlug` in `DomainsStep`.
 			theme: dependencies.themeSlugWithRepo || themeSlugWithRepo,
+			// `options.vertical` will be deprecated in favour of `options.site_vertical`
 			vertical: siteVertical || undefined,
 			siteGoals: siteGoals || undefined,
 			site_style: siteStyle || undefined,
 			site_information: siteInformation || undefined,
+			// `options.siteType` will be deprecated in favour of `options.site_segment`
 			siteType: siteType || undefined,
+			site_segment: getSiteTypePropertyValue( 'slug', siteType, 'id' ) || undefined,
+			site_segment_slug: siteType || undefined,
```

Comment:
Mentioned this in the backend diff, but it'd be better to handle the slug lookups there, since the mobile app will only pass the IDs. Sorry for not making that clear in the card.